beacon-chain/slasher/chunks.go

```diff
@@ -252,29 +263,42 @@ func (m *MaxSpanChunksSlice) CheckSlashable(
 		)
 	}
 
-	if targetEpoch < maxTarget {
-		existingAttRecord, err := slasherDB.AttestationRecordForValidator(
-			ctx, validatorIdx, maxTarget,
-		)
-		if err != nil {
-			return nil, errors.Wrapf(
-				err, "could not get existing attestation record at target %d", maxTarget,
-			)
-		}
+	if targetEpoch >= maxTarget {
+		// The incoming attestation is not surrounded by any existing ones.
+		return nil, nil
+	}
 
-		if existingAttRecord == nil {
-			return nil, nil
-		}
+	// The incoming attestation is surrounded by an existing one.
+	existingAttRecord, err := slasherDB.AttestationRecordForValidator(ctx, validatorIdx, maxTarget)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not get existing attestation record at target %d", maxTarget)
+	}
 
-		if existingAttRecord.IndexedAttestation.Data.Source.Epoch < sourceEpoch {
-			surroundedVotesTotal.Inc()
-			return &ethpb.AttesterSlashing{
-				Attestation_1: existingAttRecord.IndexedAttestation,
-				Attestation_2: attestation.IndexedAttestation,
-			}, nil
-		}
+	if existingAttRecord == nil {
+		// This case should normally not happen. If this happen, it means we previously
+		// recorded in our min/max DB an distance corresponding to an attestaiton, but WITHOUT
+		// recording the attestation itself. As a consequence, we say there is no surrounded vote,
+		// but we log an error.
+		log.Errorf("No existing attestation record found for validator %d at target %d, while a surrounded vote was detected.",
+			validatorIdx, maxTarget,
+		)
```

Comment:
Avoid templated log messages, it makes it harder to monitor / aggregate these.

Prefer using logrus fields https://pkg.go.dev/github.com/sirupsen/logrus#Logger.WithFields

Reply:
Fixed in this commit: https://github.com/prysmaticlabs/prysm/pull/13589/commits/bd02048eec6dace73b384f77ae1a31ce14070a61